src/maxdiffusion/models/train.py

```diff
@@ -491,32 +491,43 @@ def compute_loss(unet_params):
     # for checkpointing
     eval_checkpoints = []
     start_time = time.time()
+    last_log_step_num = 0
     for step in np.arange(start_step, config.max_train_steps):
         example_batch = load_next_batch(data_iterator, example_batch, config)
         unet_state, train_metric, train_rngs = p_train_step(unet_state,
                                                             example_batch,
                                                             train_rngs)
-
+        
         step_num = step + 1
         samples_count = total_train_batch_size * step_num
 
         if config.write_metrics and (step_num % config.metrics_period == 0 or step_num == config.max_train_steps):
             new_time = datetime.datetime.now()
+            
             step_time_delta = new_time - last_step_completion
-            # using global vars _buffered_step, _buffered_metrics
-            if _buffered_step is None:
-                step_num_delta = step_num
-                _buffered_step_num = step_num
-            else:
-                step_num_delta = step - _buffered_step
-                _buffered_step_num = _buffered_step + 1
-            _buffered_sample_count = total_train_batch_size * _buffered_step_num
+            step_num_delta = step_num - last_log_step_num
+
             # record metrics of current period
             record_scalar_metrics(train_metric, step_time_delta, step_num_delta, per_device_tflops, learning_rate_scheduler(step))
-            # print metrics of previous period
-            mllog_utils.maybe_train_step_log(config, start_step, _buffered_step_num, _buffered_sample_count, _buffered_metrics)
-            write_metrics(writer, local_metrics_file, running_gcs_metrics, train_metric, step, config)
+
+            if config.metrics_buffer:
```

Comment:
while mllog for this branch mlperf  branch will always turned on. we can use write_metric here to make write metrics out as optional.

Reply:
Now leaving only one flag, write_metrics for using buffer. The default behavior if no buffer with mllog on